artificial_algorand_contract/trade/trade_contract.py

```diff
@@ -0,0 +1,224 @@
+from pyteal import (
+    And,
+    App,
+    Assert,
+    Bytes,
+    Concat,
+    Cond,
+    Ed25519Verify,
+    Global,
+    Gtxn,
+    If,
+    Int,
+    Mode,
+    OnComplete,
+    Return,
+    Seq,
+    compileTeal,
+)
+
+from ..classes.algorand import TealCmdList, TealPackage, TealParam
+from .asset_config import AssetConfig, aUSD_ID
+
+TEAL_VERSION = 5
+USD_ATOM_IN_ONE = Int(10**6)
+
+
+def gen_teal_param(asset_config: AssetConfig) -> TealParam:
+    """
+    Generate the teal param by AssetConfig
+    Args:
+        asset_config (AssetConfig): config of asset, can be found in asset_config.py
+    Returns:
+        TealParam: teal param
+    """
+    # for typing and readability
+
+    """ Smart contract typing """
+    cmd_list: TealCmdList = asset_config["contract_cmd_list"]
+    local_ints_scheme = asset_config["contract_local_ints_scheme"]
+    local_bytes_scheme = asset_config["contract_local_bytes_scheme"]
+    global_ints_scheme = asset_config["contract_global_ints_scheme"]
+    global_bytes_scheme = asset_config["contract_global_bytes_scheme"]
+
+    """ Dynamic Param from typing data """
+    teal_param: TealParam = {
+        "local_ints": len(local_ints_scheme),
+        "local_bytes": len(local_bytes_scheme),
+        "global_ints": len(global_ints_scheme),
+        "global_bytes": len(global_bytes_scheme),
```

Comment:
are these parameters needed? I thought they're same for all assets

Reply:
This is for the literal typing feature in the future. all four like `global_bytes_scheme` are of python `dict` or `list` type, and they have a `__len__` prop.